pkg/tracing/sampler/sampling.go

```diff
@@ -1,39 +1,37 @@
 package sampler
 
 import (
+	"encoding/binary"
 	"fmt"
+	"math"
 
 	sdktrace "go.opentelemetry.io/otel/sdk/trace"
 	"go.opentelemetry.io/otel/trace"
 )
 
-type randGenerator interface {
-	Float64() float64
-}
-
 type RandomRatioBased struct {
 	sdktrace.Sampler
-	rnd      randGenerator
-	fraction float64
+	max uint64
 }
 
 // NewRandomRatioBased creates a sampler based on random number.
 // fraction parameter should be between 0 and 1 where:
 // fraction >= 1 it will always sample
 // fraction <= 0 it will never sample
-func NewRandomRatioBased(fraction float64, rnd randGenerator) sdktrace.Sampler {
+func NewRandomRatioBased(fraction float64) sdktrace.Sampler {
 	if fraction >= 1 {
 		return sdktrace.AlwaysSample()
 	} else if fraction <= 0 {
 		return sdktrace.NeverSample()
 	}
 
-	return &RandomRatioBased{rnd: rnd, fraction: fraction}
+	return &RandomRatioBased{max: uint64(fraction * math.MaxUint64)}
 }
 
 func (s *RandomRatioBased) ShouldSample(p sdktrace.SamplingParameters) sdktrace.SamplingResult {
+	val := binary.BigEndian.Uint64(p.TraceID[8:16])
```

Comment:
Why [8:16] btw? Is it a special part in the trace ID?

Reply:
I added the documentation, see if make sense!

I think this code should be on:

https://github.com/open-telemetry/opentelemetry-go-contrib/tree/main/propagators/aws/xray

I will try to contribute back after.